### PR TITLE
Api response rework

### PR DESCRIPTION
**Pull request by qwertyppp11**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/qwertyppp11/online-strategic-game/pull/31" target="_blank">ReviewNinja

File: api/core/Base.php [hunk 1]
```diff
@@ -11,15 +11,12 @@ class Base extends ActionContoller {
...
+        $data = $this->db->custom(
             "SELECT `base` FROM `bases` WHERE `id` = ?",
             array( $_SESSION['id'] )
         );
-        return array(
-            'status' => 'success',
-            'statusmessage' => 'success',
-            'data' => json_decode( $data['0']['base'], true )
+        return $this->utils->success(
+            json_decode( $data['0']['base'], true )
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Base.php [hunk 2]
```diff
@@ -32,8 +29,7 @@ protected function build ( $options )  {
         }
         $name = (string)$options['name'];
 
-        $updater = new UtilUpdateBaseResources();
-        $updater->update( $_SESSION['id'] );
+        $this->utils->updateBaseResources( $_SESSION['id'] );
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Base.php [hunk 3]
```diff
@@ -46,22 +42,20 @@ protected function build ( $options )  {
...
-            return array(
-                'status' => 'error',
-                'statusmessage' => 'unknown'
+            return $this->utils->error(
+                'unknown building'
             );
         }
 
-        $data = $db->custom( "SELECT `resources`, `base`, `tech` FROM `bases` WHERE `id` = ?", array($_SESSION['id']) );
+        $data = $this->db->custom( "SELECT `resources`, `base`, `tech` FROM `bases` WHERE `id` = ?", array($_SESSION['id']) );
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found
* Line exceeds 120 characters; contains 126 characters

File: api/core/Base.php [hunk 4]
```diff
@@ -70,57 +64,44 @@ protected function build ( $options )  {
         $new_building['name'] = $name;
         $new_building['level'] = 1;
         array_push( $base, $new_building );
-        $db->custom_no_return( "UPDATE `bases` SET `resources` = ?, `base` = ? WHERE `id` = ?",
+        $this->db->custom_no_return( "UPDATE `bases` SET `resources` = ?, `base` = ? WHERE `id` = ?",
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Opening parenthesis of a multi-line function call must be the last content on the line

File: api/core/ActionContoller.php [hunk 5]
```diff
@@ -5,6 +5,14 @@
     protected $allowed_actions = array ();
     protected $allowed_actions_with_no_login = array ();
 
+    protected $utils;
+    protected $db;
```
Comment (qwertyppp11) [inline api/core/ActionContoller.php]: __[phpmd]__
* Avoid variables with short names like $db. Configured minimum length is 3.

File: api/core/Base.php [hunk 6]
```diff
@@ -70,57 +64,44 @@ protected function build ( $options )  {
...
+            return $this->utils->error(
+                'no-count'
             );
         }
 
         $count = intval($options['count']);
 
-        $db = new db();
-        $base = $db->custom("SELECT `base` FROM `bases` WHERE `id` = ?", array($_SESSION['id']) );
+        $base = $this->db->custom("SELECT `base` FROM `bases` WHERE `id` = ?", array($_SESSION['id']) );
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Expected 0 spaces before closing bracket; 1 found

File: api/core/ActionContoller.php [hunk 7]
```diff
@@ -5,6 +5,14 @@
...
+
+    public function __construct ( $utils, $db ) {
```
Comment (qwertyppp11) [inline api/core/ActionContoller.php]: __[phpmd]__
* Avoid variables with short names like $db. Configured minimum length is 3.

File: api/core/Base.php [hunk 8]
```diff
@@ -70,57 +64,44 @@ protected function build ( $options )  {
...
         $base = json_decode( $base[0]['base'], true );
 
         array_splice( $base, $count, 1 );
 
-        $db->custom_no_return( "UPDATE `bases` SET `base` = ? WHERE `id` = ?",
+        $this->db->custom_no_return( "UPDATE `bases` SET `base` = ? WHERE `id` = ?",
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Opening parenthesis of a multi-line function call must be the last content on the line

File: api/core/ApiContoller.php [hunk 9]
```diff
@@ -25,7 +25,9 @@ public function processRequest ( $type, $action, $options ) {
             );
         }
 
-        $controller = new $type();
+        $utils = new Utils();
+        $db = new db();
```
Comment (qwertyppp11) [inline api/core/ApiContoller.php]: __[phpmd]__
* Avoid variables with short names like $db. Configured minimum length is 3.

File: api/core/Base.php [hunk 10]
```diff
@@ -70,57 +64,44 @@ protected function build ( $options )  {
...
     }
 
     protected function getResources () {
-        $updater = new UtilUpdateBaseResources;
-        return array(
-            'status' => 'success',
-            'statusmessage' => 'success',
-            'data' => $updater->update( $_SESSION['id'] )
+        return $this->utils->success(
+            $this->utils->updateBaseResources( $_SESSION['id'] )
```
Comment (qwertyppp11) [inline api/core/Base.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/System.php [hunk 11]
```diff
@@ -70,73 +65,73 @@ protected function register ( $options ) {
...
             $uname = htmlspecialchars($uname);
 
             //пользователь
-            $db->custom_no_return( "INSERT INTO `users` (`id`, `uname`, `upass`, `umail`, `reg_time`) VALUES (NULL, ?, ?, ?, ?)",
+            $this->db->custom_no_return( "INSERT INTO `users` (`id`, `uname`, `upass`, `umail`, `reg_time`) VALUES (NULL, ?, ?, ?, ?)",
                 array( $uname, password_hash($options['upass'], PASSWORD_DEFAULT), $options['umail'], time() )
             );
-            $id = $db->custom( "SELECT `id` FROM `users` WHERE `uname` LIKE ?",
-                array( $uname )
+            $id = $this->db->custom(
```
Comment (qwertyppp11) [inline api/core/System.php]: __[phpmd]__
* Avoid variables with short names like $id. Configured minimum length is 3.

File: api/core/System.php [hunk 12]
```diff
@@ -70,73 +65,73 @@ protected function register ( $options ) {
...
-        $db = new db();
-
-        if (empty( $db->custom("SELECT `id` FROM `users` WHERE `uname` LIKE ?",array($options['uname'])) )){
+        $existing_user = $this->db->custom(
+            "SELECT `id` FROM `users` WHERE `uname` LIKE ?",
+            array(
+                $options['uname']
+            )
+        );
+        if ( empty($existing_user) ){
```
Comment (qwertyppp11) [inline api/core/System.php]: __[code_sniffer]__
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found
* Expected 1 space after closing parenthesis; found 0
* Blank line found at start of control structure

File: api/core/Utils.php [hunk 13]
```diff
@@ -1,8 +1,46 @@
 <?php
 
-class UtilUpdateBaseResources {
+class Utils {
+
+    public function generateBaseCoordinates () {
+        $data = array();
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* Avoid variables with short names like $x. Configured minimum length is 3.

File: api/core/System.php [hunk 14]
```diff
@@ -70,73 +65,73 @@ protected function register ( $options ) {
...
 
             $uname = trim($options['uname']);
             $uname = stripcslashes($uname);
             $uname = htmlspecialchars($uname);
 
             //пользователь
-            $db->custom_no_return( "INSERT INTO `users` (`id`, `uname`, `upass`, `umail`, `reg_time`) VALUES (NULL, ?, ?, ?, ?)",
+            $this->db->custom_no_return( "INSERT INTO `users` (`id`, `uname`, `upass`, `umail`, `reg_time`) VALUES (NULL, ?, ?, ?, ?)",
```
Comment (qwertyppp11) [inline api/core/System.php]: __[code_sniffer]__
* Opening parenthesis of a multi-line function call must be the last content on the line
* Line exceeds 120 characters; contains 135 characters

File: api/core/Utils.php [hunk 15]
```diff
@@ -1,8 +1,46 @@
...
+        $x = 0;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* Avoid variables with short names like $y. Configured minimum length is 3.

File: api/core/System.php [hunk 16]
```diff
@@ -149,40 +144,32 @@ protected function register ( $options ) {
...
     protected function unlogin ( $options ) {
-
         $_SESSION['id'] = 0;
         $_SESSION['uname'] = '';
-        return array (
-            'status' => 'success',
-            'statusmessage' => 'success'
-        );
-
+        return $this->utils->success ();
```
Comment (qwertyppp11) [inline api/core/System.php]: __[code_sniffer]__
* Space before opening parenthesis of function call prohibited

File: api/core/Utils.php [hunk 17]
```diff
@@ -1,8 +1,46 @@
...
+        $y = 0;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* Avoid variables with short names like $db. Configured minimum length is 3.

File: api/core/ActionContoller.php (hunk 7 above)
Comment (qwertyppp11) [inline api/core/ActionContoller.php]: __[code_sniffer]__
* Expected "function abc(...)"; found "function abc (...)"
* Expected 0 spaces before opening parenthesis; 1 found
* Expected 0 spaces between opening bracket and argument "$utils"; 1 found
* Expected 0 spaces between argument "$db" and closing bracket; 1 found
* Opening brace should be on a new line

File: api/core/Utils.php [hunk 18]
```diff
@@ -1,8 +1,46 @@
...
+            global $x, $y;
+            $x = rand( -46000, 46000);
+            $y = rand( -46000, 46000);
+            $data = $db->custom(
+                ' SELECT `id` FROM `bases` WHERE `x` = ? AND `y` = ? ',
+                array(
+                    $x,
+                    $y
+                )
+            );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* The method generateBaseCoordinates() utilizes a goto statement.

File: api/core/Utils.php [hunk 19]
```diff
@@ -1,8 +1,46 @@
...
+        if ( !empty($data) ) goto back;
+        return array(
+            'x' => $x,
+            'y' => $y
+        );
+    }
+
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* updateBaseResources accesses the super-global variable $_SESSION.
* Avoid variables with short names like $id. Configured minimum length is 3.

File: api/core/Tech.php [hunk 20]
```diff
@@ -10,29 +10,27 @@ class Tech extends ActionContoller {
...
-            'statusmessage' => 'success',
-            'data' => json_decode( $data, true )
+        $data = $this->db->custom(
+            "SELECT `tech` FROM `bases` WHERE `id` = ?",
+            array(
+                $_SESSION['id']
+            )
+        )[0]['tech'];
+        return $this->utils->success(
+            json_decode( $data, true )
```
Comment (qwertyppp11) [inline api/core/Tech.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 21]
```diff
@@ -1,8 +1,46 @@
...
+    public function updateBaseResources ( $id ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* Avoid variables with short names like $db. Configured minimum length is 3.

File: api/core/Tech.php [hunk 22]
```diff
@@ -43,28 +41,26 @@ protected function learn ( $options ) {
...
+            return $this->utils->error(
+                'unknown tech'
             );
         } else {
             $tech = $tech_tree[$tier][$branch];
         }
 
-        $updater = new UtilUpdateBaseResources();
-        $res = $updater->update( $_SESSION['id'] );
+        $res = $this->utils->updateBaseResources( $_SESSION['id'] );
```
Comment (qwertyppp11) [inline api/core/Tech.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 23]
```diff
@@ -1,8 +1,46 @@
...
+        $data = $db->custom("SELECT `resources`, `base`, `tech` FROM `bases` WHERE `id` = ?", array($id) );
+        $resources = json_decode( $data[0]['resources'], true );
+        $base = json_decode( $data[0]['base'], true );
+        $tech = json_decode( $data[0]['tech'], true );
+        $resources = $this->count_new_resources( $resources, $base, $tech );
+        $db->custom_no_return( "UPDATE `bases` SET `resources` = ? WHERE `id` = ?",
+            array( json_encode($resources), $_SESSION['id'] )
+        );
+        return $resources;
+    }
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* count_new_resources accesses the super-global variable $_SERVER.
* The method count_new_resources is not named in camelCase.
* The variable $building_value is not named in camelCase.
* The variable $income_key is not named in camelCase.
* The variable $income_value is not named in camelCase.
* The variable $resources_per_second is not named in camelCase.
* The variable $resources_unit_time is not named in camelCase.
* Avoid unused parameters such as '$tech'.

File: api/core/Tech.php [hunk 24]
```diff
@@ -43,28 +41,26 @@ protected function learn ( $options ) {
...
 
-        $db = new db();
-        $tech_user = $db->custom(
+        $tech_user = $this->db->custom(
             "SELECT `tech` FROM `bases` WHERE `id` = ?",
-            array( $_SESSION['id'])
+            array(
```
Comment (qwertyppp11) [inline api/core/Tech.php]: __[code_sniffer]__
* Whitespace found at end of line

File: api/core/Utils.php [hunk 25]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* The method count_new_resources uses an else expression. Else is never necessary and you can simplify the code to work without else.

File: api/core/Utils.php [hunk 26]
```diff
@@ -1,8 +1,46 @@
 <?php
 
-class UtilUpdateBaseResources {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Each class must be in a namespace of at least one level (a top-level vendor name)
* Opening brace of a class must be on the line after the definition

File: api/core/Utils.php [hunk 27]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
+                'message' => 'success'
+            );
+        } else {
+            return array (
+                'status' => 'success',
+                'message' => 'success',
+                'data' => $data
+            );
+        }
+    }
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[phpmd]__
* The method success uses an else expression. Else is never necessary and you can simplify the code to work without else.

File: api/core/Utils.php [hunk 28]
```diff
@@ -1,8 +1,46 @@
 <?php
 
-class UtilUpdateBaseResources {
+class Utils {
+
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected "function abc(...)"; found "function abc (...)"
* Expected 0 spaces before opening parenthesis; 1 found
* Opening brace should be on a new line

File: api/core/Utils.php [hunk 29]
```diff
@@ -1,8 +1,46 @@
...
+    public function generateBaseCoordinates () {
+        $data = array();
+        $x = 0;
+        $y = 0;
+        $db = new db();
+        back:
+            global $x, $y;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited

File: api/core/Utils.php [hunk 30]
```diff
@@ -1,8 +1,46 @@
...
+            $x = rand( -46000, 46000);
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited

File: api/core/Utils.php (hunk 18 above)
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Line indented incorrectly; expected at least 12 spaces, found 8
* Inline control structures are not allowed
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 31]
```diff
@@ -1,8 +1,46 @@
...
+            $y = rand( -46000, 46000);
+            $data = $db->custom(
+                ' SELECT `id` FROM `bases` WHERE `x` = ? AND `y` = ? ',
+                array(
+                    $x,
+                    $y
+                )
+            );
+        if ( !empty($data) ) goto back;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Line indented incorrectly; expected at least 12 spaces, found 8

File: api/core/Utils.php [hunk 32]
```diff
@@ -1,8 +1,46 @@
...
+        return array(
+            'x' => $x,
+            'y' => $y
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Line indented incorrectly; expected at least 12 spaces, found 8

File: api/core/Utils.php (hunk 19 above)
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected "function abc(...)"; found "function abc (...)"
* Expected 0 spaces before opening parenthesis; 1 found
* Expected 0 spaces between opening bracket and argument "$id"; 1 found
* Expected 0 spaces between argument "$id" and closing bracket; 1 found
* Opening brace should be on a new line

File: api/core/Utils.php [hunk 33]
```diff
@@ -1,8 +1,46 @@
...
+        );
+    }
+
+    public function updateBaseResources ( $id ) {
+        $db = new db;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 34]
```diff
@@ -1,8 +1,46 @@
...
+        $data = $db->custom("SELECT `resources`, `base`, `tech` FROM `bases` WHERE `id` = ?", array($id) );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 35]
```diff
@@ -1,8 +1,46 @@
...
+        $resources = json_decode( $data[0]['resources'], true );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 36]
```diff
@@ -1,8 +1,46 @@
...
+        $base = json_decode( $data[0]['base'], true );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 37]
```diff
@@ -1,8 +1,46 @@
...
+        $tech = json_decode( $data[0]['tech'], true );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 38]
```diff
@@ -1,8 +1,46 @@
...
+        $resources = $this->count_new_resources( $resources, $base, $tech );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Opening parenthesis of a multi-line function call must be the last content on the line

File: api/core/Utils.php (hunk 23 above)
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Method name "Utils::count_new_resources" is not in camel caps format
* Expected "function abc(...)"; found "function abc (...)"
* Expected 0 spaces before opening parenthesis; 1 found
* Expected 0 spaces between opening bracket and argument "$resources"; 1 found
* Expected 0 spaces between argument "$tech" and closing bracket; 1 found
* Opening brace should be on a new line

File: api/core/Utils.php [hunk 39]
```diff
@@ -1,8 +1,46 @@
...
+        $db->custom_no_return( "UPDATE `bases` SET `resources` = ? WHERE `id` = ?",
+            array( json_encode($resources), $_SESSION['id'] )
+        );
+        return $resources;
+    }
 
     private function count_new_resources ( $resources, $base, $tech ) {
+        $config = new Config();
+        $buildings = $config->BUILDINGS;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space found after opening bracket of FOREACH loop
* Space found before closing bracket of FOREACH loop
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 40]
```diff
@@ -1,8 +1,46 @@
...
         foreach ( $base as $value ) {
             //поиск постройки
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space found after opening bracket of FOREACH loop
* Space found before closing bracket of FOREACH loop
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 41]
```diff
@@ -1,8 +1,46 @@
...
             foreach ( $buildings as $building_value ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 42]
```diff
@@ -27,7 +65,6 @@ private function count_new_resources ( $resources, $base, $tech ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 43]
```diff
@@ -27,7 +65,6 @@ private function count_new_resources ( $resources, $base, $tech ) {
                 }
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space found after opening bracket of FOREACH loop
* Space found before closing bracket of FOREACH loop
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 44]
```diff
@@ -27,7 +65,6 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
             }
         }
-
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 45]
```diff
@@ -27,7 +65,6 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
         foreach ( $resources as $key => $value ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 46]
```diff
@@ -27,7 +65,6 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
             if ( !isset($value['update']) ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Line exceeds 120 characters; contains 132 characters

File: api/core/Utils.php [hunk 47]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
         return $resources;
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 48]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
     }
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 49]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
 
-    public function update ( $id ) {
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Line exceeds 120 characters; contains 130 characters

File: api/core/Utils.php [hunk 50]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
-
-        $db = new db;
-        $data = $db->custom("SELECT `resources`, `base`, `tech` FROM `bases` WHERE `id` = ?", array($id) );
-
-        $config = new Config();
-        $buildings = $config->BUILDINGS;
-        $resources = json_decode( $data[0]['resources'], true );
-        $base = json_decode( $data[0]['base'], true );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space found after opening bracket of FOREACH loop
* Space found before closing bracket of FOREACH loop
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 51]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
-        $tech = json_decode( $data[0]['tech'], true );
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 52]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
+    public function success ( $data = null ) {
+        if ( $data == null ) {
+            return array (
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 53]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
+                'status' => 'success',
+                'message' => 'success'
+            );
+        } else {
+            return array (
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected "function abc(...)"; found "function abc (...)"
* Expected 0 spaces before opening parenthesis; 1 found
* Expected 0 spaces between opening bracket and argument "$data"; 1 found
* Expected 0 spaces between argument "$data" and closing bracket; 1 found
* Opening brace should be on a new line

File: api/core/Utils.php [hunk 54]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
+                'status' => 'success',
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found

File: api/core/Utils.php [hunk 55]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
+    }
 
-        $db->custom_no_return( "UPDATE `bases` SET `resources` = ? WHERE `id` = ?",
-            array( json_encode($resources), $_SESSION['id'] )
+    public function error ( $message = '' ) {
+        return array (
+            'status' => 'error',
+            'message' => $message
         );
-
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* Expected "function abc(...)"; found "function abc (...)"
* Expected 0 spaces before opening parenthesis; 1 found
* Expected 0 spaces between opening bracket and argument "$message"; 1 found
* Expected 0 spaces between argument "$message" and closing bracket; 1 found
* Opening brace should be on a new line

File: api/core/Utils.php [hunk 56]
```diff
@@ -37,24 +74,28 @@ private function count_new_resources ( $resources, $base, $tech ) {
...
-        return $resources;
-
     }
+
 }
 
 ?>
```
Comment (qwertyppp11) [inline api/core/Utils.php]: __[code_sniffer]__
* The closing brace for the class must go on the next line after the body

**Comment by qwertyppp11** [comment]
Click here if you want to create code_sniffer's auto-correct Pull Request 
[](https://sideci.com/github_repositories/59024557/auto_correct/code_sniffer/pull_request/31/branch/api-response-rework)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-code_sniffer)

**Pull request merged**